frontend/src/lib/components/DangerousDeleteModal.svelte

```diff
@@ -0,0 +1,121 @@
+<svelte:options runes={true} />
+
+<script lang="ts">
+import { i18n } from "$lib/i18n.svelte";
+import { alertStore } from "$lib/stores/alertStore.svelte";
+import { Button, Input, Label, Modal, Spinner } from "flowbite-svelte";
+import { CheckCircleOutline } from "flowbite-svelte-icons";
+import ExclamationCircleOutline from "flowbite-svelte-icons/ExclamationCircleOutline.svelte";
+import { onMount } from "svelte";
+
+type DeletionWillAffectTotals = {
+	[key: string]: number;
+};
+
+type DryRunnableDataResponse =
+	| {
+			data: { deletion_executed: boolean; error?: never; would_delete?: never };
+	  }
+	| {
+			data: {
+				would_delete: DeletionWillAffectTotals;
+				error?: never;
+				deletion_executed?: never;
+			};
+	  }
+	| { error: string };
+
+let {
+	open = $bindable(false),
+	deleteDryRunnableRequest,
+	afterDelete,
+	intendedConfirmCode,
+}: {
+	open: boolean;
+	deleteDryRunnableRequest: (
+		dryRun: boolean,
+	) => Promise<DryRunnableDataResponse>;
+	afterDelete: () => void;
+	intendedConfirmCode: string;
+} = $props();
+
+let deletionWillAffectTotals = $state<DeletionWillAffectTotals>({});
+let deleteConfirmCode: string = $state("");
+let deleteDone: boolean = $state(false);
+
+let sendDeleteRequest = async () => {
+	if (deleteConfirmCode === intendedConfirmCode) {
+		const { data, error } = await deleteDryRunnableRequest(false);
+		console.error("Error on real call:", error);
+
+		if (data.deletion_executed) {
+			deleteDone = true;
+			afterDelete(); // refresh the list items this is in etc.
+		} else {
+			alertStore.showAlert(i18n.tr.admin.deleteError, error, true, false);
+			console.error(error);
+		}
+	}
+};
+
+$effect(async () => {
+	if (open) {
+		const { data, error } = await deleteDryRunnableRequest(true);
+		if (error) {
+			alertStore.showAlert(i18n.tr.admin.deleteError, "", true, false);
+			console.error(error);
+		}
+		if (data.would_delete) {
+			deletionWillAffectTotals = data.would_delete;
+		}
+	} else {
+		deletionWillAffectTotals = {};
+		deleteDone = false;
+	}
```

Comment:
should we also clear the `deleteConfirmCode` here to avoid it being pre-filled next time the dialog is used?